pkg/tlsutils/tlsconfig.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tlsutils
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+// CreateTLSConfig creates tls.Config instance from TLS parameters passed in environment variables with the given prefix
+func CreateTLSConfig(prefix string) (*tls.Config, error) {
+	caFile := os.Getenv(fmt.Sprintf("%s_CA_FILE", prefix))
+	certFile := os.Getenv(fmt.Sprintf("%s_CERT_FILE", prefix))
+	keyFile := os.Getenv(fmt.Sprintf("%s_KEY_FILE", prefix))
+	serverName := os.Getenv(fmt.Sprintf("%s_TLS_SERVER_NAME", prefix))
+	isInsecureStr := strings.ToLower(os.Getenv(fmt.Sprintf("%s_TLS_INSECURE", prefix)))
```

Comment:
Please correct me if I'm wrong, but I just opened the first provider in the list, which happened to be AWS Route53 and one of the first things that it does is call to `session.NewSessionWithOptions(...)` which executes code in aws-go-sdk that load environment variables like `AWS_ACCESS_KEY`, `AWS_ACCESS_TOKEN`, `AWC_CA_BUNDLE`, `AWS_REGION` and so on. This is just hidden inside SDK. And this seems logical since I didn't see `--help` printing any flags that I can use to provide my AWS credentials/region/etc. Here the picture is similar - most of designate environment variables are hidden inside gophercloud library.
But generally, I agree with you as having hidden settings that are not even some minor tweaks is not something that one can like. I don't think that it need to become external-dns CLI flags for two reasons:
1) It is likely to be too many of them. Some of them are mutually exclusive or must be provided in pairs etc. This goes beyond simple CLI parsing capabilities. Without refactoring to existing code, it will turn into a function that knows all the settings of all providers and validates if the flags were provided in valid configuration which defeats the whole purpose of provider separations. And it makes help screen just to complex  
2) For some of parameters (secrets, passwords) it is generally a bad idea to have them in commandline. For the same reasons that you cannot specify password in ssh CLI flag. On the other hand, having them in environment variables makes it easy to automatically populate them from k8s Secrets/ConfigMaps.
That's why I think that `--help --provider name` is the best solution here. But it goes beyond the scope of this PR and need to be implemented consistently for all providers